handle below use-case and print in the log etails about invalid jira ticket

Instead of huge stacktrace I'f like to see information about JIRA_ID etc with valid information to be able to easily find a mistake in code

### Error updating database.  Cause: org.postgresql.util.PSQLException: ERROR: value too long for type character varying(45)
### The error may involve defaultParameterMap
### The error occurred while setting parameters
### SQL: INSERT INTO WORK_ITEMS (JIRA_ID, HASH_CODE,              TYPE,                  BLOCKER,           DESCRIPTION, TEST_CASE_ID)    VALUES (     ?,     ?,            ?,                 ?,             ?, ?    )
### Cause: org.postgresql.util.PSQLException: ERROR: value too long for type character varying(45)
; SQL []; ERROR: value too long for type character varying(45); nested exception is org.postgresql.util.PSQLException: ERROR: value too long for type character varying(45)
        at org.springframework.jdbc.support.SQLStateSQLExceptionTranslator.doTranslate(SQLStateSQLExceptionTranslator.java:102) ~[spring-jdbc-4.3.20.RELEASE.jar:4.3.20.RELEASE]
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:73) ~[spring-jdbc-4.3.20.RELEASE.jar:4.3.20.RELEASE]
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:82) ~[spring-jdbc-4.3.20.RELEASE.jar:4.3.20.RELEASE]
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:82) ~[spring-jdbc-4.3.20.RELEASE.jar:4.3.20.RELEASE]
        at org.mybatis.spring.MyBatisExceptionTranslator.translateExceptionIfPossible(MyBatisExceptionTranslator.java:75) ~[mybatis-spring-1.3.0.jar:1.3.0]
        at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:447) ~[mybatis-spring-1.3.0.jar:1.3.0]
        at com.sun.proxy.$Proxy27.insert(Unknown Source) ~[na:na]
        at org.mybatis.spring.SqlSessionTemplate.insert(SqlSessionTemplate.java:279) ~[mybatis-spring-1.3.0.jar:1.3.0]
        at org.apache.ibatis.binding.MapperMethod.execute(MapperMethod.java:46) ~[mybatis-3.2.3.jar:3.2.3]
        at org.apache.ibatis.binding.MapperProxy.invoke(MapperProxy.java:43) ~[mybatis-3.2.3.jar:3.2.3]
        at com.sun.proxy.$Proxy51.createWorkItem(Unknown Source) ~[na:na]
        at com.qaprosoft.zafira.services.services.application.WorkItemService.createWorkItem(WorkItemService.java:38) ~[zafira-services-2.0.jar:na]
        ... 101 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: value too long for type character varying(45)
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440) ~[postgresql-42.2.4.jar:42.2.4]
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183) ~[postgresql-42.2.4.jar:42.2.4]
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308) ~[postgresql-42.2.4.jar:42.2.4]
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441) ~[postgresql-42.2.4.jar:42.2.4]
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365) ~[postgresql-42.2.4.jar:42.2.4]
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:150) ~[postgresql-42.2.4.jar:42.2.4]
        at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:139) ~[postgresql-42.2.4.jar:42.2.4]
        at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.execute(NewProxyPreparedStatement.java:989) ~[c3p0-0.9.1.1.jar:0.9.1.1]
        at org.apache.ibatis.executor.statement.PreparedStatementHandler.update(PreparedStatementHandler.java:41) ~[mybatis-3.2.3.jar:3.2.3]
        at org.apache.ibatis.executor.statement.RoutingStatementHandler.update(RoutingStatementHandler.java:66) ~[mybatis-3.2.3.jar:3.2.3]
        at org.apache.ibatis.executor.SimpleExecutor.doUpdate(SimpleExecutor.java:45) ~[mybatis-3.2.3.jar:3.2.3]
        at org.apache.ibatis.executor.BaseExecutor.update(BaseExecutor.java:100) ~[mybatis-3.2.3.jar:3.2.3]
        at org.apache.ibatis.executor.CachingExecutor.update(CachingExecutor.java:75) ~[mybatis-3.2.3.jar:3.2.3]
        at org.apache.ibatis.session.defaults.DefaultSqlSession.update(DefaultSqlSession.java:148) ~[mybatis-3.2.3.jar:3.2.3]
        at org.apache.ibatis.session.defaults.DefaultSqlSession.insert(DefaultSqlSession.java:137) ~[mybatis-3.2.3.jar:3.2.3]
        at sun.reflect.GeneratedMethodAccessor715.invoke(Unknown Source) ~[na:na]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_181]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_181]
        at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:434) ~[mybatis-spring-1.3.0.jar:1.3.0]

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestService.java
@@ -146,13 +146,13 @@ public class TestService {
     private void validateTestFieldsLength(Test test) {
         String errorMessage = "";
         if (is255SymbolsLengthExceeded(test.getName())) {
-            errorMessage += "name, ";
+            errorMessage += "name(" + test.getName().length() + ") , ";
         }
         if (is255SymbolsLengthExceeded(test.getTestGroup())){
-            errorMessage += "testGroup, ";
+            errorMessage += "testGroup("+ test.getTestGroup().length() +"), ";
         }
         if (is255SymbolsLengthExceeded(test.getDependsOnMethods())){
-            errorMessage += "dependsOnMethods";
+            errorMessage += "dependsOnMethods("+ test.getDependsOnMethods().length() +")";
         }
         if(StringUtils.isNotEmpty(errorMessage)){
             errorMessage = "Test ID: "+ test.getId() + ", Test name: "+ test.getName() + "\nFields exceeding 255 symbols restriction: " + errorMessage;

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestService.java
@@ -155,7 +155,7 @@ public class TestService {
             errorMessage += "dependsOnMethods";
         }
         if(StringUtils.isNotEmpty(errorMessage)){
-            errorMessage = "Test ID: "+ test.getId() + "Test name: "+ test.getName() + "\nFields exceeding 255 symbols restriction: " + errorMessage;
+            errorMessage = "Test ID: "+ test.getId() + ", Test name: "+ test.getName() + "\nFields exceeding 255 symbols restriction: " + errorMessage;
             LOGGER.error(errorMessage);
         }
     }

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestService.java
@@ -146,16 +146,16 @@ public class TestService {
     private void validateTestFieldsLength(Test test) {
         String errorMessage = "";
         if (is255SymbolsLengthExceeded(test.getName())) {
-            errorMessage += "name\n";
+            errorMessage += "name, ";
         }
         if (is255SymbolsLengthExceeded(test.getTestGroup())){
-            errorMessage += "testGroup\n";
+            errorMessage += "testGroup, ";
         }
         if (is255SymbolsLengthExceeded(test.getDependsOnMethods())){
-            errorMessage += "dependsOnMethods\n";
+            errorMessage += "dependsOnMethods";
         }
         if(StringUtils.isNotEmpty(errorMessage)){
-            errorMessage = "Test ID: "+ test.getId() + "\nFields exceeding 255 symbols restriction:\n" + errorMessage;
+            errorMessage = "Test ID: "+ test.getId() + "Test name: "+ test.getName() + "\nFields exceeding 255 symbols restriction: " + errorMessage;
             LOGGER.error(errorMessage);
         }
     }

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestService.java
@@ -15,13 +15,10 @@
  *******************************************************************************/
 package com.qaprosoft.zafira.services.services.application;
 
-import static com.qaprosoft.zafira.models.dto.TestRunStatistics.Action.MARK_AS_BLOCKER;
-import static com.qaprosoft.zafira.models.dto.TestRunStatistics.Action.MARK_AS_KNOWN_ISSUE;
-import static com.qaprosoft.zafira.models.dto.TestRunStatistics.Action.REMOVE_BLOCKER;
-
-import java.util.*;
-import java.util.stream.Collectors;
-
+import com.qaprosoft.zafira.dbaccess.dao.mysql.application.TestMapper;
+import com.qaprosoft.zafira.dbaccess.dao.mysql.application.search.SearchResult;
+import com.qaprosoft.zafira.dbaccess.dao.mysql.application.search.TestCaseSearchCriteria;
+import com.qaprosoft.zafira.dbaccess.dao.mysql.application.search.TestSearchCriteria;
 import com.qaprosoft.zafira.models.db.Status;
 import com.qaprosoft.zafira.models.db.Tag;
 import com.qaprosoft.zafira.models.db.Test;
@@ -30,6 +27,12 @@ import com.qaprosoft.zafira.models.db.TestCase;
 import com.qaprosoft.zafira.models.db.TestConfig;
 import com.qaprosoft.zafira.models.db.TestRun;
 import com.qaprosoft.zafira.models.db.WorkItem;
+import com.qaprosoft.zafira.models.db.WorkItem.Type;
+import com.qaprosoft.zafira.models.dto.TestRunStatistics;
+import com.qaprosoft.zafira.services.exceptions.ServiceException;
+import com.qaprosoft.zafira.services.exceptions.TestNotFoundException;
+import com.qaprosoft.zafira.services.services.application.integration.impl.JiraService;
+import net.rcarz.jiraclient.Issue;
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.math.RandomUtils;
@@ -39,17 +42,21 @@ import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import com.qaprosoft.zafira.dbaccess.dao.mysql.application.TestMapper;
-import com.qaprosoft.zafira.dbaccess.dao.mysql.application.search.SearchResult;
-import com.qaprosoft.zafira.dbaccess.dao.mysql.application.search.TestCaseSearchCriteria;
-import com.qaprosoft.zafira.dbaccess.dao.mysql.application.search.TestSearchCriteria;
-import com.qaprosoft.zafira.models.db.WorkItem.Type;
-import com.qaprosoft.zafira.models.dto.TestRunStatistics;
-import com.qaprosoft.zafira.services.exceptions.ServiceException;
-import com.qaprosoft.zafira.services.exceptions.TestNotFoundException;
-import com.qaprosoft.zafira.services.services.application.integration.impl.JiraService;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
 
-import net.rcarz.jiraclient.Issue;
+import static com.qaprosoft.zafira.models.dto.TestRunStatistics.Action.MARK_AS_BLOCKER;
+import static com.qaprosoft.zafira.models.dto.TestRunStatistics.Action.MARK_AS_KNOWN_ISSUE;
+import static com.qaprosoft.zafira.models.dto.TestRunStatistics.Action.REMOVE_BLOCKER;
 
 @Service
 public class TestService {
@@ -132,7 +139,29 @@ public class TestService {
 
     @Transactional(rollbackFor = Exception.class)
     public void createTest(Test test) {
+        validateTestFieldsLength(test);
         testMapper.createTest(test);
+    }
+
+    private void validateTestFieldsLength(Test test) {
+        String errorMessage = "";
+        if (is255SymbolsLengthExceeded(test.getName())) {
+            errorMessage += "name\n";
+        }
+        if (is255SymbolsLengthExceeded(test.getTestGroup())){
+            errorMessage += "testGroup\n";
+        }
+        if (is255SymbolsLengthExceeded(test.getDependsOnMethods())){
+            errorMessage += "dependsOnMethods\n";
+        }
+        if(StringUtils.isNotEmpty(errorMessage)){
+            errorMessage = "Test ID: "+ test.getId() + "\nFields exceeding 255 symbols restriction:\n" + errorMessage;
+            LOGGER.error(errorMessage);
+        }
+    }
+
+    private boolean is255SymbolsLengthExceeded(String value) {
+        return StringUtils.isNotEmpty(value) && value.length() > 255;
     }
 
     @Transactional(rollbackFor = Exception.class)
@@ -288,6 +317,7 @@ public class TestService {
 
     @Transactional(rollbackFor = Exception.class)
     public Test updateTest(Test test) {
+        validateTestFieldsLength(test);
         testMapper.updateTest(test);
         return test;
     }

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/WorkItemService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/WorkItemService.java
@@ -15,24 +15,29 @@
  *******************************************************************************/
 package com.qaprosoft.zafira.services.services.application;
 
-import java.util.List;
-
+import com.qaprosoft.zafira.dbaccess.dao.mysql.application.WorkItemMapper;
+import com.qaprosoft.zafira.models.db.WorkItem;
+import com.qaprosoft.zafira.models.db.WorkItem.Type;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import com.qaprosoft.zafira.dbaccess.dao.mysql.application.WorkItemMapper;
-import com.qaprosoft.zafira.models.db.WorkItem;
-import com.qaprosoft.zafira.models.db.WorkItem.Type;
+import java.util.List;
 
 @Service
 public class WorkItemService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkItemService.class);
 
     @Autowired
     private WorkItemMapper workItemMapper;
 
     @Transactional(rollbackFor = Exception.class)
     public void createWorkItem(WorkItem workItem) {
+        validateWorkItemFieldsLength(workItem);
         workItemMapper.createWorkItem(workItem);
     }
 
@@ -58,6 +63,7 @@ public class WorkItemService {
 
     @Transactional(rollbackFor = Exception.class)
     public WorkItem updateWorkItem(WorkItem workItem) {
+        validateWorkItemFieldsLength(workItem);
         workItemMapper.updateWorkItem(workItem);
         return workItem;
     }
@@ -76,5 +82,20 @@ public class WorkItemService {
         } else {
             return workItem;
         }
+    }
+
+    private void validateWorkItemFieldsLength(WorkItem workItem) {
+        String errorMessage = "";
+        if (is45SymbolsLengthExceeded(workItem.getJiraId())) {
+            errorMessage += "jiraId\n";
+        }
+        if(StringUtils.isNotEmpty(errorMessage)){
+            errorMessage = "WorkItem ID: "+ workItem.getId() + "\nFields exceeding 45 symbols restriction:\n" + errorMessage;
+            LOGGER.error(errorMessage);
+        }
+    }
+
+    private boolean is45SymbolsLengthExceeded(String value) {
+        return StringUtils.isNotEmpty(value) && value.length() > 45;
     }
 }

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/WorkItemService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/WorkItemService.java
@@ -87,7 +87,7 @@ public class WorkItemService {
     private void validateWorkItemFieldsLength(WorkItem workItem) {
         String errorMessage = "";
         if (is45SymbolsLengthExceeded(workItem.getJiraId())) {
-            errorMessage += "jiraId";
+            errorMessage += "jiraId("+ workItem.getJiraId() +")";
         }
         if(StringUtils.isNotEmpty(errorMessage)){
             errorMessage = "WorkItem ID: "+ workItem.getId() + ", WorkItem JiraId: "+ workItem.getJiraId() + "\nFields exceeding 45 symbols restriction: " + errorMessage;

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/WorkItemService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/WorkItemService.java
@@ -90,7 +90,7 @@ public class WorkItemService {
             errorMessage += "jiraId";
         }
         if(StringUtils.isNotEmpty(errorMessage)){
-            errorMessage = "WorkItem ID: "+ workItem.getId() + "WorkItem JiraId: "+ workItem.getJiraId() + "\nFields exceeding 45 symbols restriction: " + errorMessage;
+            errorMessage = "WorkItem ID: "+ workItem.getId() + ", WorkItem JiraId: "+ workItem.getJiraId() + "\nFields exceeding 45 symbols restriction: " + errorMessage;
             LOGGER.error(errorMessage);
         }
     }

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/WorkItemService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/WorkItemService.java
@@ -87,10 +87,10 @@ public class WorkItemService {
     private void validateWorkItemFieldsLength(WorkItem workItem) {
         String errorMessage = "";
         if (is45SymbolsLengthExceeded(workItem.getJiraId())) {
-            errorMessage += "jiraId\n";
+            errorMessage += "jiraId";
         }
         if(StringUtils.isNotEmpty(errorMessage)){
-            errorMessage = "WorkItem ID: "+ workItem.getId() + "\nFields exceeding 45 symbols restriction:\n" + errorMessage;
+            errorMessage = "WorkItem ID: "+ workItem.getId() + "WorkItem JiraId: "+ workItem.getJiraId() + "\nFields exceeding 45 symbols restriction: " + errorMessage;
             LOGGER.error(errorMessage);
         }
     }

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/integration/impl/JenkinsService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/integration/impl/JenkinsService.java
@@ -130,6 +130,7 @@ public class JenkinsService extends AbstractIntegration<JenkinsContext> {
         } else {
             jobUrl = formatScannerJobUrl(jenkinsFolder, jenkinsHost);
         }
+        LOGGER.error("Jenkins job url: " + jobUrl);
         return jobUrl;
     }
 

--- a/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/integration/impl/JenkinsService.java
+++ b/sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/integration/impl/JenkinsService.java
@@ -113,6 +113,8 @@ public class JenkinsService extends AbstractIntegration<JenkinsContext> {
 
     public JobResult buildScannerJob(String repositoryName, Map<String, String> jobParameters, boolean rescan) {
         String jobUrl = buildJobUrl(repositoryName, rescan);
+        LOGGER.error("Jenkins job url: " + jobUrl);
+        LOGGER.error("Job parameters: " + jobParameters);
         return buildJob(jobUrl, jobParameters);
     }
 
@@ -130,7 +132,6 @@ public class JenkinsService extends AbstractIntegration<JenkinsContext> {
         } else {
             jobUrl = formatScannerJobUrl(jenkinsFolder, jenkinsHost);
         }
-        LOGGER.error("Jenkins job url: " + jobUrl);
         return jobUrl;
     }
 

--- a/sources/zafira-ws/src/main/java/com/qaprosoft/zafira/ws/controller/application/TestsAPIController.java
+++ b/sources/zafira-ws/src/main/java/com/qaprosoft/zafira/ws/controller/application/TestsAPIController.java
@@ -101,7 +101,7 @@ public class TestsAPIController extends AbstractController {
     @ApiOperation(value = "Start test", nickname = "startTest", httpMethod = "POST", response = TestType.class)
     @ApiImplicitParams({ @ApiImplicitParam(name = "Authorization", paramType = "header") })
     @PostMapping()
-    public TestType startTest(@Valid @RequestBody TestType t, @RequestHeader(value = "Project", required = false) String project)
+    public TestType startTest(@Valid @RequestBody TestType t)
             {
         Test test = testService.startTest(mapper.map(t, Test.class), t.getWorkItems(), t.getConfigXML());
         websocketTemplate.convertAndSend(

